app/models/user.rb

```diff
@@ -29,10 +30,15 @@ class User < ApplicationRecord
   scope :with_pending_or_replaced_docs, -> { self.joins(:labels).where(labels:
                                            { key: 'document', value: ['pending', 'replaced'], scope: 'private' }) }
 
-  before_validation :assign_uid
+  before_validation :assign_uid, :downcase_nickname
```

Comment:
why you need both before save amd before validation
Maybe before validation should just work

Reply:
Ah, OK. Let me do this.